### PR TITLE
define use_cuda in dropout backward to allow peephole optimization to…

### PR DESCRIPTION
**Pull request by ngimel**
… work

### CODE REVIEW

**Comment by ailzhang** [comment]
@ngimel Do you mind adding a test to prevent regression in the future? Thanks!

**Comment by ngimel** [comment]
Done, test won't pass until either #19988 or #20269 is merged.

**Comment by ngimel** [comment]
@ailzhang is it ok to be merged?

File: test/test_jit_fuser.py [hunk 1]
```diff
@@ -282,6 +282,22 @@ def funcOptMax(a, b):
...
+    @unittest.skipIf(IS_WINDOWS, "NYI: fuser support for Windows")
+    @unittest.skipIf(not RUN_CUDA, "fuser requires CUDA")
+    @skipIfRocm
+    def test_dropout(self):
+        def func(x):
+            x = torch.nn.functional.dropout(x)
+            return torch.nn.functional.relu(x)
+
+        a = torch.randn(4, 4, dtype=torch.float, device='cuda', requires_grad=True)
+        s = torch.jit.script(func, (a,))
```
Comment (fmassa) [inline test/test_jit_fuser.py]: Do the `torch.jit.script` take an example input as argument? I thought it was only trace that did.

Comment (ngimel) [reply test/test_jit_fuser.py]: It does not have to, but it can.

File: test/test_jit_fuser.py [hunk 2]
```diff
@@ -282,6 +282,22 @@ def funcOptMax(a, b):
...
+        self.assertAllFused(s.graph_for(a,), except_for={'aten::div', 'prim::Constant'})
```
Comment (wanchaol) [inline test/test_jit_fuser.py]: `aten::div` is a fusable op right? why it does not get fused?

Comment (ngimel) [reply test/test_jit_fuser.py]: It's division of a scalar argument, no point in fusing.

**Review by wanchaol — APPROVED** [review]
